### PR TITLE
RM33177 - cache the users map also for issues

### PR DESCRIPTION
**Pull request by daniela-mateescu**

### CODE REVIEW

File: lib/application_controller_patch.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module ApplicationControllerPatch
+    def self.included(base)
+      base.send(:include, InstanceMethods)
+      base.send(:before_action, :init_users_cache, only: [:show])
```
Comment (mircea-negreanu) [inline lib/application_controller_patch.rb:4]: I'm guessing that this is the base `Controller` class and we should add this hash only in `IssuesControlle` and `ForumController`.

What do you think?

File: app/helpers/advanced_messenger_helper.rb [hunk 2]
```diff
@@ -74,29 +74,26 @@ def getUnreadNotificationsForCurrentUserCount()
     def getUsersReadStatus(read_by_users)
         read_by_users = JSON.parse(read_by_users)
         read_statuses = Hash.new
-        # For the moment the global users cache is implemented only for forum messages
-        # There is an open issue to implement it also for issues notes: https://redmine.xops-online.com/issues/33177
-        users_cache = @users_cache != nil ? @users_cache : Hash.new
         read_by_users.keys.each_with_index do |user_id, index|
```
Comment (mircea-negreanu) [inline app/helpers/advanced_messenger_helper.rb:77]: I'm not sure I see where you make the difference between `forum` messages and `issues`.

I mean if this is a common code, that how did it work for forum but not for notes?

Comment (daniela-mateescu) [reply app/helpers/advanced_messenger_helper.rb:77]: Inainte: pentru forum: @users_cache != nil ca il instantiam inainte, insa pentru issue, cum nu era instantiat se folosea un map local in aceasta functie instantiat cu Hash.new

File: lib/application_controller_patch.rb (hunk 1 above)
Comment (daniela-mateescu) [reply lib/application_controller_patch.rb:4]: It worked like you suggested

**Pull request merged**